src/utils/Catalyst/index.ts

```diff
@@ -3,22 +3,31 @@ import isEthereumAddress from 'validator/lib/isEthereumAddress'
 
 import { isSameAddress } from '../../entities/Snapshot/utils'
 
-import { Avatar, ProfileResponse } from './types'
+import { CatalystProfile, DclProfile, ProfileResponse } from './types'
 
 const CATALYST_URL = 'https://peer.decentraland.org'
 export const DEFAULT_AVATAR_IMAGE = 'https://decentraland.org/images/male.png'
 
-export async function getProfile(address: string): Promise<Avatar | null> {
+function getDclProfile(profile: CatalystProfile | null, address: string): DclProfile {
+  const profileHasName = !!profile && profile.hasClaimedName && !!profile.name && profile.name.length > 0
+  const username = profileHasName ? profile.name : null
+  const hasAvatar = !!profile && !!profile.avatar
+  const avatar = hasAvatar ? profile.avatar.snapshots.face256 : DEFAULT_AVATAR_IMAGE
+  return { username, avatar, hasCustomAvatar: hasAvatar, address: address.toLowerCase() }
+}
+
+export async function getProfile(address: string): Promise<DclProfile> {
   if (!isEthereumAddress(address)) {
     throw new Error(`Invalid address provided. Value: ${address}`)
   }
 
   const response: ProfileResponse = await (await fetch(`${CATALYST_URL}/lambdas/profile/${address}`)).json()
-
-  return response.avatars.length > 0 ? response.avatars[0] : null
+  const profile = response.avatars.length > 0 ? response.avatars[0] : null
+  return getDclProfile(profile, address)
 }
 
-export async function getProfiles(addresses: string[]): Promise<(Avatar | null)[]> {
+// TODO: this could be cached per address
```

Comment:
I'd remove this comment as this is the Catalyst client API. We should let the cache decision to whoever want to use it.